homeassistant/components/deconz.py

```diff
@@ -0,0 +1,198 @@
+    Load light data containing which lights are available.
+    Load sensor data containing which sensors are available.
+    Start websocket for push notification of state changes from Deconz.
+    """
+    from pydeconz import DeconzSession
+    deconz = DeconzSession(hass.loop, **deconz_config)
+    hass.data[DECONZ_DATA] = deconz
+    result = yield from deconz.load_parameters()
+    if result is False:
+        _LOGGER.error('Failed to setup deCONZ component')
+        return False
+    hass.async_add_job(discovery.async_load_platform(hass,
+                                                     'light',
+                                                     DOMAIN,
+                                                     {},
+                                                     config))
+    hass.async_add_job(discovery.async_load_platform(hass,
+                                                     'binary_sensor',
+                                                     DOMAIN,
+                                                     {},
+                                                     config))
+    hass.async_add_job(discovery.async_load_platform(hass,
+                                                     'scene',
+                                                     DOMAIN,
+                                                     {},
+                                                     config))
+    hass.async_add_job(discovery.async_load_platform(hass,
+                                                     'sensor',
+                                                     DOMAIN,
+                                                     deconz_config,
+                                                     config))
+    deconz.start()
+    return True
+
+
+def request_configuration(hass, config, deconz_config):
+    """Request configuration steps from the user."""
+    configurator = hass.components.configurator
+
+    # pylint: disable=unused-argument
+    @asyncio.coroutine
+    def configuration_callback(data):
+        """Set up actions to do when our configuration callback is called."""
+        from pydeconz.utils import get_api_key
+        api_key = yield from get_api_key(hass.loop, **deconz_config)
+        if api_key:
+            deconz_config[CONF_API_KEY] = api_key
+            result = yield from _setup_deconz(hass, config, deconz_config)
+            save_json(hass.config.path(CONFIG_FILE), {CONF_API_KEY: api_key})
+            if result:
+                hass.async_add_job(configurator.request_done, request_id)
+                return True
+        hass.async_add_job(
+            configurator.notify_errors, request_id, "Didn't get an API key.")
+        return False
+
+    request_id = configurator.request_config(
+        "deCONZ", configuration_callback,
+        description="deCONZ -> Menu -> Settings -> Unlock Gateway",
+        submit_caption="I have unlocked the gateway",
+        link_name='deCONZ platform documentation',
```

Comment:
`link_name` and `link_url` are deprecated. `description` can now be markdown with as many links as you want.

Reply:
I would suggest adding a link to `http://<deCONZ IP>/edit_system.html` which leads directly to the page with the 'Unlock Gateway' button.

Reply:
Moved to markdown